### ISSUE TITLE
Quantity * Measurement = Quantity; M*Q = M

### ISSUE THREAD

**Issue by mankoff** [hgrecco/pint#278]
If a measurement is the 2nd term in an equation, the result is a quantity, not a measurement:

``` python
x = (ureg.m).plus_minus(1)
y = ureg.m
type(x*y)      # <class 'pint.unit.build_measurement_class.<locals>.Measurement'>
type(y*x)      # <class 'pint.unit.build_quantity_class.<locals>.Quantity'>
```

**Comment by hgrecco** [comment]
We need to fix this!

**Comment by alexfeld** [comment]
Hi. What is the status on this issue? The latest version of pint still appears to have this behavior.

**Comment by alexfeld** [comment]
Hi. What is the status on this issue? The latest version of pint still appears to have this behavior.

I have added a hack to the Quantity class that handles the add, sub, mul, div, and trudiv functions that checks the type of the 'other' input and changes the return based on whether the 'other' is a measurement or not and whether 'self' is a measurement or not. The new `__sub__` method is below:

    def __add__(self, other):
        if isinstance(other, self._REGISTRY.Measurement) and not isinstance(self, self._REGISTRY.Measurement):
            return self._REGISTRY.Measurement.__add__(other, self)
        else:
            return self._add_sub(other, operator.add)  # old method

This works for now, but it would be nice to see if there is a better solution. I did check this new version of Quantity with the provided testing suite and everything seemed to work.
